src/main/java/hudson/plugins/validating_string_parameter/ValidatingStringParameterValue.java

```diff
@@ -73,13 +76,27 @@ public BuildWrapper createBuildWrapper(AbstractBuild<?, ?> build) {
                 public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
                     throw new AbortException("Invalid value for parameter [" + getName() + "] specified: " + value);
                 }
+
+                @Override
+                public Launcher decorateLauncher(AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException, Run.RunnerAbortedException {
+                    throw new AbortException("Invalid value for parameter [" + getName() + "] specified: " + value);
+                }
             };
         } else {
             return null;
         }
     }
 
     @Override
+    public void buildEnvironment(Run<?, ?> build, EnvVars env) {
```

Comment:
It does not seem to be the best approach. Maybe makes sense to use `QueueDecisionHandler` or something like that